rewrite-java/src/main/java/org/openrewrite/java/cleanup/SimplifyBooleanExpressionVisitor.java

```diff
@@ -83,6 +87,16 @@ public J visitBinary(J.Binary binary, P p) {
             }
         }
 
+        getCursor().dropParentUntil(J.class::isInstance).putMessage(MAYBE_AUTO_FORMAT_ME, "");
+        return j;
+    }
+
+    @Override
+    public @Nullable J postVisit(J tree, P p) {
+        J j = super.postVisit(tree, p);
+        if (getCursor().pollMessage(MAYBE_AUTO_FORMAT_ME) != null) {
+            j = new AutoFormatVisitor<>().visit(j, p, getCursor());
+        }
         return j;
```

Comment:
Note how we aren't using `maybeAutoFormat` here, we're just blatantly forcing it by virtue of having the cursor message... doesn't feel quite right.

Need more time to digest it, but also-- isn't the input param of `tree` always going to be the same...? Would it be a conceptual _faux pax_ to have `postVisit` receive the result of the element's first `preVisit` call separately in order to determine referential changes?

Putting a pin in this. Just including the changes as an example for the moment.

Reply:
hmm, I wonder if you should cursor.putMessage the J.Binary (or the parent J, not clear to me), to use as the before in postVisit, so you could use maybeAutoFormat.

Reply:
should just be `maybeAutoFormat(tree, j, getCursor())` unless i'm missing something. no need for any data to be on the cursor message. the thing that's being formatted is this tree, and not one of the descendents.

Reply:
wouldn't this result in autoformatting every parent tree from the tree that's changed and "upwards"? if we override `postVisit` to say, "for each tree, if any subtrees below me have changed, run autoformat on me and return that result", then wouldn't every parent upward have `autoformat` ran on it too in each respective `postVisit` due to any one child changing?